gst/nnstreamer/tensor_query/nnstreamer_edge_internal.c

```diff
@@ -477,12 +494,16 @@ nns_edge_release_handle (nns_edge_h edge_h)
         ("Failed to release the edge handle due to disconnection failure.");
     return NNS_EDGE_ERROR_CONNECTION_FAILURE;
```

Comment:
need unlock here.

Reply:
I added new function to close connections to avoid dead-lock. Please see the changes again.